charon/src/bin/charon-driver/translate/translate_crate_to_ullbc.rs

```diff
@@ -358,6 +359,28 @@ impl<'tcx, 'ctx> TranslateCtx<'tcx, 'ctx> {
         }
         Ok(self.translated.get_item(id).unwrap())
     }
+
+    /// Return a map from source file name to file content.
+    fn read_source_files(&mut self) -> HashMap<FileName, String> {
+        // Retrieve the source map
+        let source_map = self.tcx.sess.source_map();
+
+        // Read all the files
+        use std::ops::Deref;
+        source_map
+            .files()
```

Comment:
Will that include every file of every dependency? Can you check if the impact on json size is still small? Otherwise we should only add to the map whenever we register a new file id

Reply:
Yes, it probably loads all the files. But if you look carefully, you'll notice that when I compute the map from file id to file content, I only lookup the ids which we register in the spans, meaning that only the relevant files are written to the JSON.
Also, I thought of loading the file contents whenever registering a new file id (it seemed the natural way of doing) but actually it is cumbersome because we have access to *hax* information, not rustc information, meaning we actually don't know how to retrieve the file content. As a consequence, it proved more natural to load all the files, and compute an intermediate mapping from (LLBC) file name to file content.

Reply:
Then maybe we should let hax do this for us, or at least we could keep the `Arc<String>` instead of cloning everything

Reply:
Either way please add a comment explaining this

Reply:
Yes, makes sense: I kept the `Arc<String>`